doc/using_pytorch.rst

```diff
@@ -566,94 +566,127 @@ The function should return a byte array of data serialized to content_type.
+    |           |--model.pth
+    |
+    |           code
+    |               |--inference.py
+    |               |--requirements.txt
+
+Where ``requirments.txt`` is an optional file that specifies dependencies on third-party libraries.
+
+With this file structure, run the following command to package your model as a ``tar.gz`` file:
+
+``tar -czf model.tar.gz my_model code``
+
+Upload model.tar.gz to S3
+-------------------------
+
+**Note**: This step is optional. The ``PyTorchModel`` object will upload your model file to S3.
+You can specify a bucket location by providing a value for the ``code_location`` parameter when you create the ``PyTorchModel`` object,
+or you can use the default S3 session bucket.
+
+After you package your model into a ``tar.gz`` file, upload it to an S3 bucket by running the following Python code:
 
 .. code:: python
 
-    my_training_job_name = 'MyAwesomePyTorchTrainingJob'
-    pytorch_estimator = PyTorch.attach(my_training_job_name)
+    import boto3
+    import sagemaker
+    s3 = boto3.client('s3')
 
-After attaching, if the training job has finished with job status "Completed", it can be
-``deploy``\ ed to create a SageMaker Endpoint and return a
-``Predictor``. If the training job is in progress,
-attach will block and display log messages from the training job, until the training job completes.
+    from sagemaker.s3 import S3Uploader
+    S3Uploader.upload('model.tar.gz', s3:://my-bucket/my-path/model.tar.gz')
 
-The ``attach`` method accepts the following arguments:
+Where ``my-bucket`` is the name of your S3 bucket, and ``my-path`` is the folder where you want to store the model.
 
--  ``training_job_name:`` The name of the training job to attach
-   to.
--  ``sagemaker_session:`` The Session used
-   to interact with SageMaker
 
-Deploy Endpoints from model data
+You can also upload to S3 by using the AWS CLI:
+
+.. code:: python
+ 
+    aws s3 cp model.tar.gz s3://my-bucket/my-path/model.tar.gz
+
+
+To run this command, you'll need to have the AWS CLI tool installed. For information about installing the AWS CLI,
+see `Installing the AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html>`_.
+
+Create a ``PyTorchModel`` object
 --------------------------------
 
-In addition to attaching to existing training jobs, you can deploy models directly from model data in S3.
-The following code sample shows how to do this, using the ``PyTorchModel`` class.
+Now call the :class:`sagemaker.pytorch.model.PyTorchModel` constructor to create a model object, and then call its ``deploy()`` method to deploy your model for inference.
 
 .. code:: python
 
-    pytorch_model = PyTorchModel(model_data='s3://bucket/model.tar.gz', role='SageMakerRole',
-                                 entry_point='transform_script.py')
+    from sagemaker import get_execution_role
+    role = get_execution_role()
+    
+    pytorch_model = PyTorchModel(model_data='s3://my-bucket/my-path/model.tar.gz', role=role,
```

Comment:
because `get_execution_role` has to be run from a notebook, I lean toward just using a placeholder string for the role instead

```suggestion
    pytorch_model = PyTorchModel(model_data='s3://my-bucket/my-path/model.tar.gz', role='my-role',
```